src/lib.rs

```diff
@@ -1947,10 +1947,7 @@ where
     /// constructs a new `Ref`. If the check fails, it returns `None`.
     #[inline]
     pub fn new_unaligned(bytes: B) -> Option<Ref<B, T>> {
-        if bytes.len() != mem::size_of::<T>() {
-            return None;
-        }
-        Some(Ref(bytes, PhantomData))
+        Ref::new(bytes)
     }
```

Comment:
Since these methods just become aliases for their non-unaligned counterparts, what's the rationale for keeping them? If it's just to minimize immediate breakage, I'd suggest that they are mark deprecated and removed from a future release.

Reply:
It's to minimize breakage, yeah. I don't want to mark them as deprecated yet for two reasons:
- Without deprecating them, this is a breaking change [1]
- I'm not convinced we'll actually want to get rid of them; there's more work to be done around how our API expresses what is and isn't validated at runtime, and I could see a future in which we keep these and have them return a different error type than the constructors that need to check alignment

I've filed https://github.com/google/zerocopy/issues/338 to track this.

[1] At some point I want to come up with a more unified story around how we keep track of "pending breaking changes". It'd be good to be able to say "this is a breaking change we'll make as soon as we're ready to release e.g. 0.8." We have that loosely tracked in issues right now, but it's neither consistent nor enforced.